datafusion/optimizer/src/common_subexpr_eliminate.rs

```diff
@@ -53,10 +53,32 @@ type ExprSet = HashMap<Identifier, (Expr, usize, DataType)>;
 /// here is not such a good choose.
 type Identifier = String;
 
-/// Perform Common Sub-expression Elimination optimization.
+/// Performs Common Sub-expression Elimination optimization.
 ///
-/// Currently only common sub-expressions within one logical plan will
+/// This optimization improves query performance by computing expressions that
+/// appear more than once and reusing those results rather than re-computing the
+/// same value
+///
+/// Currently only common sub-expressions within a single `LogicalPlan` are
 /// be eliminated.
+///
+/// # Example
+///
+/// Given a projection that computes the same expensive expression
+/// multiple times such as parsing as string as a date with `to_date` twice:
+///
+/// ```text
+/// ProjectionExec(expr=[extract (day from to_date(c1)), extract (year from to_date(c1))])
+/// ```
+///
+/// This optimization will rewrite the plan to compute the common expression once
+/// using a new `ProjectionExec` and then rewrite the original expressions to
+/// refer to that new column.
+///
+/// ```text
+/// ProjectionExec(exprs=[extract (day from new_col), extract (year from new_col)]) <-- reuse here
```

Comment:
Thanks @alamb just a question, so to read from the plan that Common Sub-expression Elimination optimization was applied we need to refer to `new_col`  but how it differs from plan with projection alias but no optimization applied?

In short my question is it possible to read from the plan that optimization has been applied?